collectors/proc.plugin/sys_class_power_supply.c

```diff
@@ -245,13 +245,18 @@ int do_sys_class_power_supply(int update_every, usec_t dt) {
                     if(unlikely(ps->capacity->fd == -1)) {
                         error("Cannot open file '%s'", ps->capacity->filename);
                         power_supply_free(ps);
+                        ps = NULL;
```

Comment:
If you do `ps = NULL;` it will crash immediately afterwards in line 252: `ssize_t r = read(ps->capacity->fd, buffer, 30);`

Reply:
Actually, I don't get that `read` at all. Even in the previous version, we should end up segfaulting when `ps->capacity->fd == -1`, right? I mean, it zeroes everything. Looks like the code is missing an `} else {` here.

Reply:
It probably did not segfault immediately due to it not being NULL. We either need to move the code in the `if (ps)` or use an `else` as you suggested.

Reply:
I decided to use an `else`, I tested both to see what would be more readable, and I confess I liked more the `else`.